dask/dataframe/io/io.py

```diff
@@ -499,21 +499,31 @@ def _link(token, result):
     return None
 
 
-def _df_to_bag(df, index=False):
+def _df_to_bag(df, index=False, format="tuple"):
     if isinstance(df, pd.DataFrame):
-        return list(map(tuple, df.itertuples(index)))
+        if format == "tuple":
+            return list(map(tuple, df.itertuples(index)))
+        elif format == "dict":
+            tuple_to_dict = lambda x: dict(x._asdict())
+            return list(map(tuple_to_dict, df.itertuples(index)))
```

Comment:
I suspect that there is a better way to do this.

Looking briefly at the Pandas API I find the `df.to_dict(orient="records")` option, which might be a better fit here.

Reply:
Hi @mrocklin, Thanks for the comments,

Actually I tried using df.to_dict(orient='records') , but if user  specfiying the index = True | False,
I changed the orient accordingly by using  `orient = "index" if index else "records"`

so `_df_to_bag` function returns following output respectively, 

**case 1 index =False :**
[{'Name': 'John', 'Age': 23, 'Fruit': 'orange'},
 {'Name': 'Lara', 'Age': 21, 'Fruit': 'apple'},
 {'Name': 'James', 'Age': 22, 'Fruit': 'apricot'},
 {'Name': 'Pablo', 'Age': 21, 'Fruit': 'cherry'}]

**case 2 index = True :**
{0: {'Name': 'John', 'Age': 23, 'Fruit': 'orange'},
 1: {'Name': 'Lara', 'Age': 21, 'Fruit': 'apple'},
 2: {'Name': 'James', 'Age': 22, 'Fruit': 'apricot'},
 3: {'Name': 'Pablo', 'Age': 21, 'Fruit': 'cherry'}}

But the output of ddf.to_bag is list of index of above function, please refer screenshot : 

New Approach: 

![image](https://user-images.githubusercontent.com/25167333/126910504-e8513b8b-7a44-49ba-8729-4b544d13a2d2.png)

Current approach:

![image](https://user-images.githubusercontent.com/25167333/126910564-dd562252-9541-44a6-aa42-7480ee7eb2f0.png)


So Please let me know, how to achieve the desired output with the newer approach (df.to_dict), 
Anything I missed here🤔 ?

Reply:
@rajagurunath `df.to_dict(orient="records")` should be sufficient for the default case when `index=False`. For the `index=True` case something like this list comprehension should incorporate the index information while still only relying on pandas' public API

```python
In [45]: df
Out[45]:
      name  age
foo   john    7
bar   mary   77
baz  sally   45

In [46]: [{**{"index": idx}, **values} for values, idx in zip(df.to_dict("records"), df.index)]
Out[46]:
[{'index': 'foo', 'name': 'john', 'age': 7},
 {'index': 'bar', 'name': 'mary', 'age': 77},
 {'index': 'baz', 'name': 'sally', 'age': 45}]
```

This works, but is somewhat verbose. There may be a better way to do this with pandas' API

Reply:
Thanks for the detailed code & comment explanation, I tried searching for some time not able to find a less verbose solution for the above use case, So implemented the above-suggested version